toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -367,7 +369,14 @@ class StreamDiagnosticConsumer : public DiagnosticConsumer {
     if (message.location.column_number > 0) {
       *stream_ << message.location.line << "\n";
       stream_->indent(message.location.column_number - 1);
-      *stream_ << "^\n";
+      if (message.location.length <= 1) {
+        *stream_ << "^";
+      } else {
+        for (int i = 0; i < message.location.length; ++i) {
+          *stream_ << "~";
+        }
```

Comment:
`message.location.length` could be longer than `message.location.line` in the case of a multi-line token. Would it make sense to end the `~`s early in that case?

Please can you also add a test for this, for example just:

```
'''
hello
'''
```

which currently produces
```
test.carbon:1:1: ERROR: Unrecognized declaration introducer.
'''
^
```
but I think after this change will produce something like
```
test.carbon:1:1: ERROR: Unrecognized declaration introducer.
'''
~~~~~~~~~~~~~
```

Reply:
I simply limited the underline to end at the end of the line but will take note that we will have to truly handle this once we add inline messages and multiple ranges.